core/src/consensus/tendermint/mod.rs

```diff
@@ -591,8 +591,8 @@ impl TendermintInner {
         let current_height = self.height;
         if current_height == height && self.view == proposal_view {
             self.proposal = Some(proposal.hash());
-            let prev_step = mem::replace(&mut self.step, TendermintState::Propose);
```

Comment:
IMO, this line was added to move `self.step` to a new variable, which is used in match expression, and fills the `self.step` with an empty or default value. However, the `TendermintState::Propose` is not a good empty or default value. Also, the code omitted to return prev_step.

So I changed it by cloning the step field to use in the match expression.